### PR TITLE
Prohibit using `nested_filter`, `nested_path` and new `nested` Option at the same time in FieldSortBuilder

### PR DESCRIPTION
**Pull request by cbuescher**
Currently we allow both "old" and "new" way of setting nested sorts on the 
FieldSortBuilder at the same time. This should throw an error, instead the user
should choose one of the two possible options.

Also adding testing for the now deprecated nestedPath/nestedFilter parameters,
inlcuding checks that they emmit warnings on parsing and that the new
NestetedSortBuilder overwrites the deprecated parameters when building the
SortField.

Relates to #17286

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java [hunk 1]
```diff
@@ -79,18 +84,23 @@ public FieldSortBuilder randomFieldSortBuilder() {
...
             builder.sortMode(randomFrom(SortMode.values()));
         }
-
         if (randomBoolean()) {
             builder.setNestedSort(createRandomNestedSort(3));
         }
-
+        // the following are alternative ways to setNestedSort for nested sorting
+        if (randomBoolean()) {
+            builder.setNestedFilter(randomNestedFilter());
```
Comment (jimczi) [inline core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java]: Not related to tests but this function should be deprecated.

File: core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java [hunk 2]
```diff
@@ -79,18 +84,23 @@ public FieldSortBuilder randomFieldSortBuilder() {
...
+        }
+        if (randomBoolean()) {
+            builder.setNestedPath(randomAlphaOfLengthBetween(1, 10));
```
Comment (jimczi) [inline core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java]: same here

File: core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java [hunk 3]
```diff
@@ -79,18 +84,23 @@ public FieldSortBuilder randomFieldSortBuilder() {
         if (randomBoolean()) {
             builder.sortMode(randomFrom(SortMode.values()));
         }
-
```
Comment (jimczi) [inline core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java]: Shouldn't we test only three cases: no_sort, new_sort, old_sort ? Mixing the old and the new format should not be allowed.

**Review by jimczi — COMMENTED** [review]
Thanks @cbuescher 
I left some comments.

File: core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java (hunk 1 above)
Comment (cbuescher) [reply core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java]: Not sure if this was omitted on purpose, but I added deprecation annotations to the setter and getter

File: core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java (hunk 2 above)
Comment (cbuescher) [reply core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java]: I added deprecation annotations to the setter and getter

File: core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java (hunk 3 above)
Comment (cbuescher) [reply core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java]: I changed this to either test one or the other. I can also add checks on the setters that prohibit setting both version at the same time, but I didn't want to change too much of the existing behaviour of the class under test in this PR, wdyt?

Comment (cbuescher) [reply core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java]: I added an additional commit to this PR that changes the current behaviour to throw an error if both formats are used at the same time. If you think this is good we can change the title of the PR to not only be about testing.

Comment (jimczi) [reply core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java]: ++ to do that

**Comment by cbuescher** [comment]
@jimczi I changed title and description to also reflect the changes in the behaviour in cases where both path/filter and the new nested option is specified. Anything else to add from your side?

**Review by jimczi — APPROVED** [review]
LGTM, thanks !

**Pull request merged**